src/core/app_irc_event.c

```diff
@@ -929,18 +929,28 @@ static void irc_event_invite(SircSession *sirc, const char *event,
     g_return_if_fail(srn_server_is_valid(srv));
     srv_user = srn_server_add_and_get_user(srv, origin);
     g_return_if_fail(srv_user);
-    chat = srn_server_add_and_get_chat(srv, origin);
-    g_return_if_fail(chat);
+
+    // Send to room chat if already open
+    chat = srn_server_get_chat(srv, chan);
+
+    // Otherwise, fall back to creating a chat with the sender
+    if (!chat) {
+        chat = srn_server_add_and_get_chat(srv, origin);
+        g_return_if_fail(chat);
+    }
+
```

Comment:
No need to "try get a `SrnChat`, otherwise add one", this is what `srn_server_add_and_get_chat` do.

Reply:
Not exactly. This code gets the *channel* chat if it already exists, and add/gets a chat to the *origin* otherwise

Reply:
Oh, I get it.